snap/naming/validate.go

```diff
@@ -332,6 +334,37 @@ func ValidateAssumes(assumes []string, currentSnapdVersion string, featureSet ma
 			}
 		}
 
+		if strings.HasPrefix(flag, "isa") {
+			// we allow keys like isa-<arch>-<isa_val>, so the result of the split will
```

Comment:
move all of this into a new function called `validateAssumedIsaArch` or something like that, to improve readability of the ValidateAssumes, and to keep things more streamlined.